File: getting_started/selection_sort.py
import random


def sort(data):
    for i in range(0, len(data) - 1):
        idx_of_min = i
        for j in range(i + 1, len(data)):
            if data[j] < data[idx_of_min]:
                idx_of_min = j

        if idx_of_min != i:
            data[i], data[idx_of_min] = data[idx_of_min], data[i]

        print(data)


if __name__ == '__main__':
    data = [random.randint(1, 20) for _ in range(20)]
    
    print("--- before ---")
    print(data)

    print("--- sort ---")
    sort(data)

    print("--- after ---")
    print(data)

File: getting_started/sort_algorithm/selection_sort.go
package sort_algorithm

import "fmt"

func SelectionSort(data []int) {
	n := len(data)

	for i := 0; i < n-1; i++ {
		idx_of_min := i
		for j := i + 1; j < n; j++ {
			if data[j] < data[idx_of_min] {
				idx_of_min = j
			}
		}

		if idx_of_min != i {
			data[i], data[idx_of_min] = data[idx_of_min], data[i]
		}

		fmt.Println(data)
	}
}
